net: tls_credentials cmake is wrong, causing build failure

**Describe the bug**
subsys/net/lib/tls_credentials folder is not building correctly because of missing link interface connections, see build output below of tests/net/lib/tls_credentials on an RT board

The issue is that the mbedTLS interface is not being linked to these sources.

**To Reproduce**
Steps to reproduce the behavior:
`west build -b mimxrt1060_evk -p always tests/net/lib/tls_credentials/`

**Expected behavior**
source files build

**Impact**
Blocking #62833

**Logs and console output**
```
-- west build: building application
[1/279] Preparing syscall dependency handling

[2/279] Generating include/generated/version.h
-- Zephyr version: 3.5.99 (/home/declan/zephyrproject/zephyr), build: zephyr-v3.5.0-1661-g256adeebd934
[58/279] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/net/lib/tls_credentials/tls_credentials_digest_raw.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/subsys/net/lib/tls_credentials/tls_credentials_digest_raw.c.obj 
ccache /home/declan/zephyr-sdk-0.16.1/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBOARD_FLASH_SIZE="CONFIG_FLASH_SIZE*1024" -DCPU_MIMXRT1062DVL6A -DKERNEL -DPICOLIBC_INTEGER_PRINTF_SCANF -DXIP_BOOT_HEADER_DCD_ENABLE=1 -DXIP_BOOT_HEADER_ENABLE=1 -DXIP_EXTERNAL_FLASH -D_FORTIFY_SOURCE=1 -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I/home/declan/zephyrproject/zephyr/kernel/include -I/home/declan/zephyrproject/zephyr/arch/arm/include -I/home/declan/zephyrproject/zephyr/drivers/memc -I/home/declan/zephyrproject/zephyr/include -I/home/declan/zephyrproject/zephyr/build/zephyr/include/generated -I/home/declan/zephyrproject/zephyr/soc/arm/nxp_imx/rt -I/home/declan/zephyrproject/zephyr/subsys/testsuite/include -I/home/declan/zephyrproject/zephyr/subsys/testsuite/ztest/include -I/home/declan/zephyrproject/zephyr/subsys/net/l2 -I/home/declan/zephyrproject/zephyr/subsys/net/lib/tls_credentials/. -I/home/declan/zephyrproject/zephyr/subsys/net/lib/dns/. -I/home/declan/zephyrproject/modules/hal/cmsis/CMSIS/Core/Include -I/home/declan/zephyrproject/zephyr/modules/cmsis/. -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1062 -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1062/drivers -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/devices/MIMXRT1062/xip -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/common -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/trng -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/lpuart -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/enet -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/igpio -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/drivers/cache/armv7-m7 -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/components/phy -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phyksz8081 -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phyar8031 -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phylan8720a -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/components/phy/device/phyrtl8211f -I/home/declan/zephyrproject/modules/hal/nxp/mcux/mcux-sdk/components/phy/mdio/enet -I/home/declan/zephyrproject/zephyr/modules/hal_nxp/. -isystem /home/declan/zephyrproject/zephyr/lib/libc/common/include -fno-strict-aliasing -Os -imacros /home/declan/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h -fno-printf-return-value -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m7 -mthumb -mabi=aapcs -mfp16-format=ieee -mtp=soft --sysroot=/home/declan/zephyr-sdk-0.16.1/arm-zephyr-eabi/arm-zephyr-eabi -imacros /home/declan/zephyrproject/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/declan/zephyrproject/zephyr/tests/net/lib/tls_credentials=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/declan/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/declan/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/subsys/net/lib/tls_credentials/tls_credentials_digest_raw.c.obj -MF zephyr/CMakeFiles/zephyr.dir/subsys/net/lib/tls_credentials/tls_credentials_digest_raw.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/subsys/net/lib/tls_credentials/tls_credentials_digest_raw.c.obj -c /home/declan/zephyrproject/zephyr/subsys/net/lib/tls_credentials/tls_credentials_digest_raw.c
In file included from <command-line>:
/home/declan/zephyrproject/zephyr/build/zephyr/include/generated/autoconf.h:217:33: fatal error: config-tls-generic.h: No such file or directory
  217 | #define CONFIG_MBEDTLS_CFG_FILE "config-tls-generic.h"
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[67/279] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/irq_init.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /home/declan/zephyrproject/zephyr/build
```

--- a/drivers/modem/CMakeLists.txt
+++ b/drivers/modem/CMakeLists.txt
@@ -12,33 +12,30 @@ zephyr_library_sources_ifdef(CONFIG_MODEM_IFACE_UART_ASYNC modem_iface_uart_asyn
 zephyr_library_sources_ifdef(CONFIG_MODEM_CMD_HANDLER modem_cmd_handler.c)
 zephyr_library_sources_ifdef(CONFIG_MODEM_SOCKET modem_socket.c)
 
+zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
+zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/lib/sockets)
+
 if(CONFIG_MODEM_UBLOX_SARA)
-	zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
 	zephyr_library_sources(ublox-sara-r4.c)
 endif()
 
 if(CONFIG_MODEM_QUECTEL_BG9X)
-	zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
 	zephyr_library_sources(quectel-bg9x.c)
 endif()
 
 if(CONFIG_MODEM_WNCM14A2A)
-	zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
 	zephyr_library_sources(wncm14a2a.c)
 endif()
 
 if(CONFIG_MODEM_GSM_PPP)
-	zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
 	zephyr_library_sources(gsm_ppp.c)
 endif()
 
 if (CONFIG_MODEM_HL7800)
-  zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
   zephyr_library_sources(hl7800.c)
 endif()
 
 if (CONFIG_MODEM_SIM7080)
-	zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
 	zephyr_library_sources(simcom-sim7080.c)
 endif()
 

--- a/drivers/wifi/eswifi/CMakeLists.txt
+++ b/drivers/wifi/eswifi/CMakeLists.txt
@@ -5,6 +5,7 @@ if(CONFIG_WIFI_ESWIFI)
   zephyr_library_include_directories(
   # IP headers
   ${ZEPHYR_BASE}/subsys/net/ip
+  ${ZEPHYR_BASE}/subsys/net/lib/sockets
   )
 
   zephyr_library_sources(

--- a/drivers/wifi/simplelink/CMakeLists.txt
+++ b/drivers/wifi/simplelink/CMakeLists.txt
@@ -3,6 +3,7 @@
 if(CONFIG_WIFI_SIMPLELINK)
   zephyr_library_include_directories(
     ${ZEPHYR_BASE}/subsys/net/lib/tls_credentials
+    ${ZEPHYR_BASE}/subsys/net/lib/sockets
     )
   zephyr_library_sources(
     simplelink_support.c

--- a/subsys/mgmt/updatehub/CMakeLists.txt
+++ b/subsys/mgmt/updatehub/CMakeLists.txt
@@ -23,3 +23,5 @@ zephyr_library_sources_ifdef(CONFIG_USERSPACE updatehub_handlers.c)
 zephyr_include_directories(
   ${ZEPHYR_BASE}/subsys/mgmt/updatehub/include
 )
+
+zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)

--- a/subsys/net/lib/sockets/CMakeLists.txt
+++ b/subsys/net/lib/sockets/CMakeLists.txt
@@ -5,37 +5,37 @@ zephyr_syscall_header(
   ${ZEPHYR_BASE}/include/zephyr/net/socket_select.h
 )
 
-zephyr_include_directories(.)
+zephyr_library_include_directories(.)
 
-zephyr_sources(
+zephyr_library_sources(
   getaddrinfo.c
   sockets.c
   sockets_select.c
 )
 
 if(NOT CONFIG_NET_SOCKETS_OFFLOAD)
-zephyr_sources(
+zephyr_library_sources(
   getnameinfo.c
   sockets_misc.c
   )
 endif()
 
-zephyr_sources_ifdef(CONFIG_NET_SOCKETS_CAN                sockets_can.c)
-zephyr_sources_ifdef(CONFIG_NET_SOCKETS_PACKET             sockets_packet.c)
-zephyr_sources_ifdef(CONFIG_NET_SOCKETS_SOCKOPT_TLS        sockets_tls.c)
-zephyr_sources_ifdef(CONFIG_NET_SOCKETS_OFFLOAD            socket_offload.c)
-zephyr_sources_ifdef(CONFIG_NET_SOCKETS_OFFLOAD_DISPATCHER socket_dispatcher.c)
-zephyr_sources_ifdef(CONFIG_NET_SOCKETS_OBJ_CORE           socket_obj_core.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETS_CAN                sockets_can.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETS_PACKET             sockets_packet.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETS_SOCKOPT_TLS        sockets_tls.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETS_OFFLOAD            socket_offload.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETS_OFFLOAD_DISPATCHER socket_dispatcher.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETS_OBJ_CORE           socket_obj_core.c)
 
 if(CONFIG_NET_SOCKETS_NET_MGMT)
-  zephyr_sources(sockets_net_mgmt.c)
-  zephyr_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
+  zephyr_library_sources(sockets_net_mgmt.c)
+  zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/ip)
 endif()
 
 if(CONFIG_SOCKS)
-  zephyr_include_directories(${ZEPHYR_BASE}/subsys/net/lib/socks)
+  zephyr_library_include_directories(${ZEPHYR_BASE}/subsys/net/lib/socks)
 endif()
 
-zephyr_sources_ifdef(CONFIG_NET_SOCKETPAIR socketpair.c)
+zephyr_library_sources_ifdef(CONFIG_NET_SOCKETPAIR socketpair.c)
 
-zephyr_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)
+zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)

--- a/subsys/net/lib/sockets/sockets.c
+++ b/subsys/net/lib/sockets/sockets.c
@@ -847,13 +847,13 @@ ssize_t zsock_sendto_ctx(struct net_context *ctx, const void *buf, size_t len,
 ssize_t z_impl_zsock_sendto(int sock, const void *buf, size_t len, int flags,
 			   const struct sockaddr *dest_addr, socklen_t addrlen)
 {
-	int ret;
+	int bytes_sent;
 
-	ret = VTABLE_CALL(sendto, sock, buf, len, flags, dest_addr, addrlen);
+	bytes_sent = VTABLE_CALL(sendto, sock, buf, len, flags, dest_addr, addrlen);
 
-	sock_obj_core_update_send_stats(sock, ret);
+	sock_obj_core_update_send_stats(sock, bytes_sent);
 
-	return ret;
+	return bytes_sent;
 }
 
 #ifdef CONFIG_USERSPACE
@@ -932,13 +932,13 @@ ssize_t zsock_sendmsg_ctx(struct net_context *ctx, const struct msghdr *msg,
 
 ssize_t z_impl_zsock_sendmsg(int sock, const struct msghdr *msg, int flags)
 {
-	int ret;
+	int bytes_sent;
 
-	ret = VTABLE_CALL(sendmsg, sock, msg, flags);
+	bytes_sent = VTABLE_CALL(sendmsg, sock, msg, flags);
 
-	sock_obj_core_update_send_stats(sock, ret);
+	sock_obj_core_update_send_stats(sock, bytes_sent);
 
-	return ret;
+	return bytes_sent;
 }
 
 #ifdef CONFIG_USERSPACE
@@ -1498,13 +1498,13 @@ ssize_t zsock_recvfrom_ctx(struct net_context *ctx, void *buf, size_t max_len,
 ssize_t z_impl_zsock_recvfrom(int sock, void *buf, size_t max_len, int flags,
 			     struct sockaddr *src_addr, socklen_t *addrlen)
 {
-	int ret;
+	int bytes_received;
 
-	ret = VTABLE_CALL(recvfrom, sock, buf, max_len, flags, src_addr, addrlen);
+	bytes_received = VTABLE_CALL(recvfrom, sock, buf, max_len, flags, src_addr, addrlen);
 
-	sock_obj_core_update_recv_stats(sock, ret);
+	sock_obj_core_update_recv_stats(sock, bytes_received);
 
-	return ret;
+	return bytes_received;
 }
 
 #ifdef CONFIG_USERSPACE

--- a/subsys/net/lib/tls_credentials/CMakeLists.txt
+++ b/subsys/net/lib/tls_credentials/CMakeLists.txt
@@ -1,15 +1,17 @@
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_include_directories(.)
+zephyr_library_include_directories(.)
 
-zephyr_sources_ifdef(CONFIG_TLS_CREDENTIALS_BACKEND_VOLATILE
+zephyr_library_sources_ifdef(CONFIG_TLS_CREDENTIALS_BACKEND_VOLATILE
   tls_credentials.c
   tls_credentials_digest_raw.c
 )
-zephyr_sources_ifdef(CONFIG_TLS_CREDENTIALS_BACKEND_PROTECTED_STORAGE
+zephyr_library_sources_ifdef(CONFIG_TLS_CREDENTIALS_BACKEND_PROTECTED_STORAGE
   tls_credentials_trusted.c
   tls_credentials_digest_raw.c
 )
-zephyr_sources_ifdef(CONFIG_TLS_CREDENTIALS_SHELL
+zephyr_library_sources_ifdef(CONFIG_TLS_CREDENTIALS_SHELL
   tls_credentials_shell.c
 )
+
+zephyr_library_link_libraries_ifdef(CONFIG_MBEDTLS mbedTLS)

